amps/ags/rm-automation/rm-automation-community-rest-api/src/test/java/org/alfresco/rest/rm/community/hold/PreventActionsOnFrozenContentV1Tests.java

```diff
@@ -0,0 +1,337 @@
+        RecordCategory categoryWithRS = getRestAPIFactory().getRecordCategoryAPI()
+            .getRecordCategory(recordFolder.getParentId());
+        dispositionScheduleService.createCategoryRetentionSchedule(categoryWithRS.getName(), false);
+        dispositionScheduleService.addCutOffImmediatelyStep(categoryWithRS.getName());
+        dispositionScheduleService.addDestroyWithGhostingImmediatelyAfterCutOff(categoryWithRS.getName());
+
+        STEP("Check the record folder has a disposition schedule");
+        RecordFolder folderWithRS = getRestAPIFactory().getRecordFolderAPI().getRecordFolder(recordFolder.getId());
+        assertNotNull(folderWithRS.getProperties().getRecordSearchDispositionAuthority());
+        assertNotNull(folderWithRS.getProperties().getRecordSearchDispositionInstructions());
+
+    }
+
+    /**
+     * Given a record category with a disposition schedule applied to records
+     * And the disposition schedule has a retain step  immediately and destroy step immediately
+     * And a complete record added to one hold
+     * When I execute the retain action
+     * Then the action is executed
+     * And the record search disposition schedule properties are updated
+     */
+    @Test
+    public void retainActionOnFrozenHeldRecords()
+    {
+        STEP("Add a category with a disposition schedule.");
+        categoryWithRS = createRootCategory(getRandomName("CategoryWithRS"));
+        dispositionScheduleService.createCategoryRetentionSchedule(categoryWithRS.getName(), true);
+        dispositionScheduleService.addRetainAfterPeriodStep(categoryWithRS.getName(), "immediately");
+        dispositionScheduleService.addDestroyWithGhostingImmediatelyAfterCutOff(categoryWithRS.getName());
+
+        STEP("Create record folder with a record.");
+        RecordCategoryChild folder = createFolder(categoryWithRS.getId(), getRandomName("RecFolder"));
+        Record record = createElectronicRecord(folder.getId(), getRandomName("elRecord"));
+        completeRecord(record.getId());
+
+        STEP("Add the record to the hold");
+        getRestAPIFactory()
+            .getHoldsAPI(getAdminUser())
+            .addChildToHold(HoldChild.builder().id(record.getId()).build(), hold.getId());
+
+        STEP("Execute the retain action");
+        rmRolesAndActionsAPI.executeAction(getAdminUser().getUsername(), getAdminUser().getPassword(), record.getName(),
+            RM_ACTIONS.END_RETENTION, null, SC_INTERNAL_SERVER_ERROR);
+
+        STEP("Check the record search disposition properties");
+        Record recordUpdated = getRestAPIFactory().getRecordsAPI().getRecord(record.getId());
+        assertTrue(recordUpdated.getProperties().getRecordSearchDispositionActionName()
+            .contains(RM_ACTIONS.END_RETENTION.getAction()));
+        assertTrue(recordUpdated.getProperties().getRecordSearchDispositionPeriod().contains("immediately"));
+    }
+
+    public Hold createHold(String parentId, Hold hold, UserModel user)
```

Comment:
Can be private & static?

Reply:
Done. It can be only static because of the non-static `RestAPIFactory`.

Reply:
I assume you meant it can only be `private` 👍